icu4c/source/i18n/number_longnames.cpp

```diff
@@ -519,10 +522,35 @@ void getCurrencyLongNameData(const Locale &locale, const CurrencyUnit &currency,
     // In ICU4J, this method gets a CurrencyData from CurrencyData.provider.
     // TODO(ICU4J): Implement this without going through CurrencyData, like in ICU4C?
     PluralTableSink sink(outArray);
+    // Here all outArray entries are bogus.
     LocalUResourceBundlePointer unitsBundle(ures_open(U_ICUDATA_CURR, locale.getName(), &status));
     if (U_FAILURE(status)) { return; }
     ures_getAllChildrenWithFallback(unitsBundle.getAlias(), "CurrencyUnitPatterns", sink, status);
     if (U_FAILURE(status)) { return; }
+    // Here the outArray[] entries are filled in with any CurrencyUnitPatterns data for locale,
+    // or if there is no CurrencyUnitPatterns data for locale since the patterns all inherited
+    // from the "other" pattern in root (which is true for many locales in CLDR 46), then only
+    // the "other" entry has a currency pattern. So now what we do is: For all valid plural keywords
+    // for the locale, if the corresponding outArray[] entry is bogus, fill it in from the "other"
+    // entry. In the longer run, clients of this should instead consider using CurrencyPluralInfo
+    // (see i18n/unicode/currpinf.h).
```

Comment:
I'm sorry; I didn't understand this at all.

Reply:
> I'm sorry; I didn't understand this at all.

The code use to iterate through the currency unit patterns to populate the set of patterns needed for all of the currency plural cases in a locale. Those currency unit patterns mostly duplicated the root "other" case which they ultimately inherit from; so now CLDR does not bother providing them if they do. So we cannot iterate over them to get the set of currency patterns for the plural cases. Instead we use PluralRules to determine what the valid plural cases are for the locale, then flesh out the currency unit patterns by copying from the "other" entry any that are missing.

Reply:
Thank you; that helps a lot.